### PR TITLE
Migrate from `react-test-renderer` to `@testing-library/react`

### PR DESCRIPTION
**Pull request by umpox**
**Migrates all of our simple React Test Renderer tests to use React Testing Library**

Part of https://github.com/sourcegraph/sourcegraph/issues/24986

This was originally going to be a series of issues to assign to GitStart, but I realised that the vast majority of these tests could be migrated with some find+replace magic.

**For now, it is intentional just to replicate the existing behavior. Over time, we want to improve these tests to test more than just snapshots**

Changes:
- Migrates all test files that just rely on `react-test-renderer` and `toJSON` to render snapshots
- Removes some unnecessary component mocks to make our snapshots closer to the actual created DOM

Benefits:
- These snapshots now output the actual created DOM that React produces. This is closer to what the user will use, and ensures that we are not testing implementation details. For example, our snapshots should not fail if we rename a component or a prop. TS can check the correct props are passed, our tests should just test what the user will actually use.
- We're close to removing `react-test-renderer`! After the follow up ticket is completed () we can remove this library and work on migrating our `enzyme` tests to `@testling-library/react`
- We get some nice coverage boosts as we mock less components

Note: **Hide whitespace changes when viewing the diff, it'll be painful otherwise!**

Current usage (prior to this PR):
image

This PR should change this quite significantly

### CODE REVIEW

**Comment by umpox** [comment]
Note: This file diff is quite large, it is mainly because React props have been converted to actual HTML attributes, e.g. `htmlFor` becomes `for` as so on. The actual changes are quite small.

**Comment by umpox** [comment]
Remaining tests will be migrated by GitStart: https://github.com/sourcegraph/sourcegraph/issues/27280

As they require a little more time to work out the correct queries to use to replicate the existing tests

**Review by valerybugakov — COMMENTED** [review]
Mamma Mia! 😯

Screenshot 2021-11-10 at 18 08 29

File: client/branded/src/components/panel/views/HierarchicalLocationsView.test.tsx [hunk 1]
```diff
@@ -1,12 +1,7 @@
-// react-visibility-sensor, used in CodeExcerpt depends on ReactDOM.findDOMNode,
-// which is not supported when using react-test-renderer + jest.
-// This mock makes it so that <VisibilitySensor /> simply becomes a <div> in the rendered output.
-jest.mock('react-visibility-sensor', () => 'VisibilitySensor')
```
Comment (valerybugakov) [inline client/branded/src/components/panel/views/HierarchicalLocationsView.test.tsx:4]: nice

File: client/web/src/enterprise/dotcom/productPlans/ProductPlanPrice.test.tsx [hunk 2]
```diff
@@ -1,80 +1,66 @@
...
+            ).asFragment()
         ).toMatchInlineSnapshot(`
-            Array [
-              "$0.08",
-              "/user/month (paid yearly)",
-            ]
+            <DocumentFragment>
+              $0.08/user/month (paid yearly)
+            </DocumentFragment>
         `)
```
Comment (valerybugakov) [inline client/web/src/enterprise/dotcom/productPlans/ProductPlanPrice.test.tsx:64]: nice [2]

**Review by valerybugakov — APPROVED** [review]
Consistency is around the corner! 🚀

One question: is it possible to remove this comment along with the redundant prop since we migrated to another renderer?

https://github.com/sourcegraph/sourcegraph/blob/bca34dc627f2668ab333f1905754705cced659f5/client/web/src/nav/GlobalNavbar.tsx#L106-L107

**Comment by umpox** [comment]
@valerybugakov Yep, looks like we don't actually use this prop anyway!

Thanks for reviewing!

**Comment by vovakulikov** [comment]
@umpox nice description of this PR with code insight! ❤️

**Pull request merged**